src/cpp/rtps/xmlparser/XMLEndpointParser.cpp

```diff
@@ -249,6 +249,38 @@ XMLP_ret XMLEndpointParser::loadXMLReaderEndpoint(tinyxml2::XMLElement* xml_endp
                 return XMLP_ret::XML_ERROR;
             }
         }
+        else if (key == TOPIC) // deprecated api
+        {
+            const char *topicName = element->Attribute(NAME);
+            const char *typeName = element->Attribute(DATA_TYPE);
+            const char *kind = element->Attribute(KIND);
+
+            rdata->topicName() = topicName ? std::string(topicName) : std::string("");
```

Comment:
This is available because of the use of internal class pointers, but I would use the method `set` rather than assign to pointer returned by `get`
```suggestion
            rdata->topicName(topicName ? std::string(topicName) : std::string(""));
```